### DISCUSSION TITLE
Want to Play Entire Local Music Folder and Limited Playlist Options

### DISCUSSION THREAD

**Discussion by johnxgg** [music-assistant/support#3192] [Feature requests and ideas]
Hi folks, I've been struggling with playing an entire music folder on my local network using Music Assistant (MA). I have a music folder mounted from my Synology NAS, which contains multiple subfolders categorized by genre or artist.

Problem 1: Adding Entire Folder to Queue or Playlist On the desktop interface, I can select all files using Ctrl + A and easily add them to the queue. However, on the Android companion app, there doesn't seem to be a simple way to add an entire folder (including its subfolders) to the queue or to a playlist. This makes it cumbersome to play large collections of music, especially when there are many files spread across different subfolders.

Problem 2: Limited Options for Multiple Selections When selecting multiple songs on the Android app, the only option available is to add them to the queue. There is no option to add the selected songs to a playlist. The "Add to Playlist" option is only available when viewing the individual song's information, making it time-consuming to create a playlist with multiple tracks.

### How to reproduce

1. Mount a music folder from Synology NAS in Music Assistant, containing multiple subfolders (e.g., categorized by genre or artist).
2. Open the Music Assistant Companion App on Android.
3. Attempt to add an entire folder (including subfolders) to the queue or a playlist.
4. Notice that there is no direct option to add the folder; you need to enter each subfolder and manually select tracks.
5. Even when multiple tracks are selected, the app only provides the option to add them to the queue, not to a playlist.

**Comment by johnxgg** [comment]
![Screenshot_20241115_231036_io homeassistant companion android](https://github.com/user-attachments/assets/7fb27ca6-9a42-449e-9a1f-dd1e2a475af7)
![Screenshot_20241115_231115_io homeassistant companion android](https://github.com/user-attachments/assets/ba26a488-50f0-4014-954c-3d8251c55f5a)

**Comment by OzGav** [comment]
This is not an issue. It is a feature request at best. MA does not revolve around folders, you should be using the library. If you want to use folders you can use some other application to create a playlist and MA can use that.

**Comment by johnxgg** [comment]
Hi OzGav, thanks for your reply. Would you be more specific that by using which app so I can create a playlist and be played by MA? I did have tried VLC's m3u but MA didn't wanna play.

**Comment by OzGav** [comment]
Others have successfully used VLC. If you are down too many levels then this might the problem https://github.com/music-assistant/hass-music-assistant/issues/2291

**Comment by OzGav** [comment]
Moving to feature request

**Comment by doctorgaul** [comment]
This would be useful for me as well.  My experience has been that playlist formats can be finicky and unreliable between platforms, so for maximum compatibility, I also have some "playlist folders."

**Comment by OzGav** [comment]
Playing a folder from the browse view is available in 2.4.0b28
